src/index.ts

```diff
@@ -73,37 +92,33 @@ export const compare = memoize((versionA: VersionInput, versionB: VersionInput):
 	versionA = normalize(versionA);
 	versionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
-	if (isAValid && !isBValid) {
-		return 1;
-	}
-	if (!isAValid && isBValid) {
-		return -1;
-	}
-	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+	const semverA = semver.parse(versionA);
+	const semverB = semver.parse(versionB);
+	if (!semverA || !semverB) {
```

Comment:
Had to change this, since otherwise TS couldn't understand that after the previous 3 `if` statements, both semvers couldn't be null. No strong opinion on this though just wanted to avoid a few `!` in the code.